### ISSUE TITLE
Alerts Directory and alerts.json Missing 

### ISSUE THREAD

**Issue by marcjbaker** [wazuh/wazuh#85]
We are attempting to utilize the Wazuh HIDS for OSSEC and ran the following as directed:
$ cd ~
$ mkdir ossec_tmp && cd ossec_tmp
$ git clone -b stable https://github.com/wazuh/wazuh.git ossec-wazuh
$ cd ossec-wazuh
$ sudo ./install.sh

After this was completed I ran the following as directed and received an error because /alerts/alerts.json was not created.

$ lsof /var/ossec/logs/alerts/alerts.json
COMMAND     PID  USER   FD   TYPE DEVICE SIZE/OFF   NODE NAME
ossec-ana 31366 ossec   10w   REG  202,0      245 274582 /var/ossec/logs/alerts/alerts.json

Please advise on steps to correct this.

**Comment by vikman90** [comment]
Hi @marcjbaker,

when you install Wazuh, the services does not start automatically, this is why the file `alerts.json` is does not yet exist.

After installing Wazuh, you may start the service using:

> `/var/ossec/bin/ossec-control start`

Then you should be able to get the alerts file.

Hope it help. If this does not solve your problem please write back to us.

Best regards.

**Comment by marcjbaker** [comment]
Thank you for your reply. The directions state that file should be in /var/ossec/logs/alerts/alerts.json. When I started the service the directory created was var/ossec/logs/alerts/2017/ and no .json files are being created. Actual file now is categorized by year and Month:

/var/ossec/logs/alerts# ls
2017  alerts.log
/var/ossec/logs/alerts# nano alerts.log
Use "fg" to return to nano.

[14]+  Stopped                 nano alerts.log
var/ossec/logs/alerts# cd ./2017/
var/ossec/logs/alerts/2017# ls
Feb  Mar
var/ossec/logs/alerts/2017# cd ./Feb/
/var/ossec/logs/alerts/2017/Feb# ls
ossec-alerts-13.log.gz   ossec-alerts-15.log.sum  ossec-alerts-18.log.gz   ossec-alerts-20.log.sum  ossec-alerts-23.log.gz   ossec-alerts-25.log.sum  ossec-alerts-28.log.sum
ossec-alerts-13.log.sum  ossec-alerts-16.log.gz   ossec-alerts-18.log.sum  ossec-alerts-21.log.gz   ossec-alerts-23.log.sum  ossec-alerts-26.log.gz
ossec-alerts-14.log.gz   ossec-alerts-16.log.sum  ossec-alerts-19.log.gz   ossec-alerts-21.log.sum  ossec-alerts-24.log.gz   ossec-alerts-26.log.sum
ossec-alerts-14.log.sum  ossec-alerts-17.log.gz   ossec-alerts-19.log.sum  ossec-alerts-22.log.gz   ossec-alerts-24.log.sum  ossec-alerts-27.log
ossec-alerts-15.log.gz   ossec-alerts-17.log.sum  ossec-alerts-20.log.gz   ossec-alerts-22.log.sum  ossec-alerts-25.log.gz   ossec-alerts-28.log.gz

Two other issues associated with this install you may be able to help us with:


1.       We uploaded the elastic-ossec-template.json using “curl -XPUT "http://localhost:9200/_template/ossec/" -d "@elastic-ossec-template.json" as directed. The expected response was received but logstash will now not start (it was running without issue prior to inputing the .json file).

2.       Kibana is generating an error “No default index pattern”. I have gone through the instructions provided by Wazuh and no mention of configuring a default index can be found.

Any assistance with these issues would be greatly appreciated.

V/r

Marc Baker

**Comment by vikman90** [comment]
Hi,

did you perform a upgrade or a clean installation? New installations should enable JSON by default. Please verify that it is son at file `/var/ossec/etc/ossec.conf`:

```
<ossec_config>
  <global>
    <jsonout_output>yes</jsonout_output>
    <!-- rest of file -->
```

Regarding Kibana, you may find some incompatibility issues if you are using Elastic/Kibana 5. Let me suggest to install the version 2.0 (master branch currently), we are doing some final tests in order to do an official release, and follow the instructions to install Logstash, Elastic and Kibana at [https://github.com/wazuh/wazuh-documentation/tree/2.0/source](https://github.com/wazuh/wazuh-documentation/tree/2.0/source).

Best regards.

**Comment by marcjbaker** [comment]
This is an upgrade. The ossec.conf file has the following for logall:

<global>
    <jsonout_output>yes</jsonout_output>
    <alerts_log>yes</alerts_log>
    <logall>no</logall>
    <logall_json>no</logall_json>
    <email_notification>no</email_notification>
    <smtp_server>smtp.example.wazuh.com</smtp_server>
    <email_from>ossecm@example.wazuh.com</email_from>
    <email_to>recipient@example.wazuh.com</email_to>
    <email_maxperhour>12</email_maxperhour>
  </global>

From your response I am assuming <logall_json>no</logall_json> should be yes, correct?


We re-ran the upgrade to ensure everything was clean and the OSSEC service is not starting. var/ossec/bin/ossec-logtest returns the following error:

2017/03/09 18:32:25 ossec-testrule: INFO: Reading decoder file etc/decoders/local_decoder.xml.
2017/03/09 18:32:25 ossec-analysisd(1231): ERROR: Invalid NULL element in the configuration.
2017/03/09 18:32:25 ossec-testrule(1202): ERROR: Configuration error at 'etc/decoders/local_decoder.xml'. Exiting.

Kibana version = 4.5.4 Build 1000

**Comment by vikman90** [comment]
No, this configuration seems to be correct: `json_output` enabled, and `logall`/`logall_json` disabled in order not to create `archives.log` or `archives.json`. File `alerts.json` should be created next time that the service start successfully.

Maybe the upgrader installed moved the local decoder but did not change the settings.

As of Wazuh v1.1.1, decoders are at `/var/ossec/etc/ossec_decoders` and `/var/ossec/etc/wazuh_decoders`. No local decoder file is created by default.

Please find a line such the following one at file `/var/ossec/etc/ossec.conf`:
> `<decoder>etc/decoders/local_decoder.xml</decoder>`

If that line exists, remove it. If you have a local decoder made by yourself, put the correct path.
Else, if you can't find that line but the error persists, create a dummy local decoder at `etc/decoders/local_decoder.xml` with this content:

> `<decoder name="local_decoder_example">`
> `  <program_name>local_decoder_example</program_name>`
> `</decoder>`

And try to start the service again.

Regards.

**Comment by marcjbaker** [comment]
Thank you for your assistance. OSSEC is now running and generating alerts. My next question is on the Kibana installation. We had all services running and then applied the 01-ossec-singlehost.conf as directed on your website. This caused the service to crash and we are unable to restart. Please provide any suggestions.

Thank you,

Marc Baker

**Comment by vikman90** [comment]
Hi,

what is the service that crashes? Could you provide logs or any clue about the problem?

**Comment by snaow** [comment]
Hi,

Let me ask you to run some commands to debug your Elastic stack situation.

Is Logstash running?

> ps aux | grep logstash

Could you check Logstash log file latest info?

> tail /var/log/logstash/.*stdout
> tail /var/log/logstash/.*stderr

Is Elasticsearch listening to localhost?

> netstat -pna | grep 9200

Do you have any indices already created?

> curl -XGET localhost:9200/_cat/ossec-*


Usually Logstash output every error on startup, please try to check it so we can help you in a better way.
You could run Logstash in foreground aslo:

> /usr/share/logstash/bin/logstash -f /etc/logstash/conf.d/01-ossec-singlehost.conf --verbose 

Thanks.

**Comment by marcjbaker** [comment]
See my answers below. Your help is greatly appreciated.

V/r

Marc Baker

**Comment by snaow** [comment]
Hi @marcjbaker,

I can't see any of the attached images on your last post.

> No – Unable to restart Elasticsearch after applying single node config on OSSEC Wazuh github
Single node configuration just consists on a Logstash configuration file, which is not related with Elasticsearch configuration, meaning Logstash can't prevent Elasticsearch to boot up.

I am sorry, I am still not sure what is exactly failing on your cluster. All the components could run individually, I mean, be sure that Logstash is up and running, then move to Elasticsearch, finally check Kibana interface.

If you can give us some logs maybe we could know what is happening, without logs, I am blind here :(

I will recommend you to perform a clean install, remove everything you have and install latest Wazuh version, which will be release in few days. The new version includes Wazuh HIDS 2.0, API 2.0, OpenSCAP, Elastic Stack 5.0 + Wazuh App.
Find a development version of the documentation at https://documentation-dev.wazuh.com/

Best regards,
Pedro.

**Comment by marcjbaker** [comment]
Pedro,

Thank you for your assistance. I have also been working with Elasticsearch and we discovered that the root cause was adding the following to /etc/Elasticsearch/Elasticsearch.yml:

Shards default number is 5 and Replicas default number is 1, if you are deploying a single-node Elastic cluster, in order to have a Green status you have to set to 1/0 shards and replicas
index.number_of_shards: 1
index.number_of_replicas: 0

The above is directly from the Wazuh-documentation website. Once these two lines were commented out, the service started without issue. If you have advice on where else this can be inserted it would be greatly appreciated.

Thank you,

Marc

**Comment by snaow** [comment]
Hi Marc,

I am glad you find your error, that is covered in the documentation:

You are right, that is covered on our documentation:
http://documentation.wazuh.com/en/latest/ossec_elk_elasticsearch.html#configuration-and-tuning

> Shards default number is 5 and Replicas default number is 1, if you are deploying a single-node Elastic cluster, in order to have a Green status you have to set to 1/0 shards and replicas

We have almost ready a new documentation, we will try to explain better those concepts.

Regarding to your last question, Elastic 2.0 version supports set shard/replica number at elasticsearch.yml and/or template file:

> [https://www.elastic.co/guide/en/elasticsearch/reference/2.1/indices-templates.html](https://www.elastic.co/guide/en/elasticsearch/reference/2.1/indices-templates.html)

Regards,
Pedro.

PS: Feel free to close the issue if you don't have any more doubts or hesitations, thanks!

**Comment by marcjbaker** [comment]
Pedro,

Thank you. One last question (thank you for your patience). I found the python script on the Wazuh github that added the OSSEC index. Kibana is now configured but not receiving data. Can you provide insite on how to configure this as I’ve yet to find it in the Wazuh documentation?

Thanks,

Marc

**Comment by snaow** [comment]
Hi,

No worries about my patient, I am happy to keep helping you, our community
is first priority :D

I am not sure what Python script are you talking about, could you provide
the link?

If you already have Elastic ossec-* indices, all you need to do is follow
the documentation at:

http://documentation.wazuh.com/en/latest/ossec_elk_kibana.html#kibana-configuration

Be sure you have already indices (you could run the _cat/indices command),
then configure Kibana index pattern.

Regards,
Pedro.

**Comment by marcjbaker** [comment]
Link is: https://github.com/wazuh/docker-ossec-elk/blob/9fb7c50e90a02e99487604362d5de3d5b24ebe96/wazuh_kibana_installer.py

I ran this script and see the indexes but no data is being forwarded from logstash or Elasticsearch and the Wazuh documentation at the link provided has no instructions on how to set this up. The logstash forwarder instructions in the wazuh documentation direct to modify configuration with:

# The list of files configurations
"files": [
{
    "paths": [
      "/var/ossec/logs/alerts/alerts.json"
     ],
    "fields": { "type": "ossec-alerts" }
}
]

There is no similar type of instruction for a single node so I’m wondering if the issue is that this needs to be done in the single server configuration as well.

V/r

Marc

**Comment by snaow** [comment]
Hi,

Are you using docker? The script is executed automatically when a docker container boot up.
If you are not using docker, I would recommend you to follow strictly the documentation, otherwise you will mess a little bit everything.

Regards,
Pedro.

**Comment by marcjbaker** [comment]
I am not using docker container but will install it. If it is vital then it seems that it should be included in the Kibana installation instructions. Overall I’ve had to do a lot of searching on the Elasticsearch user groups to get through this and wonder if there is a way to make the Wazuh directions clearer? The index and shards example from earlier is a prime example. Installing it per instructions given on your site will cause Elasticsearch to crash. I am accepting the cluster status now as yellow until another means of configuring these is found that will not crash the service. Will install docker and let you know the results.

Thank you,

Marc Baker

**Comment by snaow** [comment]
Hi,

I never said it's vital to use Docker, you could use or not, your choice.
Your Elasticsearch cluster crash because you miss two important lines about
configure your cluster, still, like I said before, we are improving it! So
it will be better, thanks for your feedback.

I bet, your cluster is in yellow because you have some indices with
replicas, cause you have just one node, you will need to disabled replicas
in all your current and further indices.

Regards,
Pedro.

**Comment by marcjbaker** [comment]
I have now installed Docker and Elasticsearch is still not forwarding logs to Kibana. As said, the cluster is in yellow because instructions to configure indices and replicas provided in Wazuh documentation caused the service to crash when followed. At this time I have not been able to find the correct directions to configure these and so am accepting the yellow status. It seems like there should be a configuration in Elasticsearch or Logstash to cause either to capture the OSSEC logs. This is shown in forwarder directions, as shown in my last email, but not in a single server configuration. If you have any advice or possible links to set up the log forwarding it would be greatly apprectiated.

V/r

Marc

**Comment by marcjbaker** [comment]
Pedro,

A config test identified the following issue:

Invalid setting for elasticsearch output plugin:

  output {
    elasticsearch {
      # This setting must be a path
      # File does not exist or cannot be opened /etc/logstash/wazuh-elastic2-template.json
      template => "/etc/logstash/wazuh-elastic2-template.json"
      ...
    }
  } {:level=>:error}

It seems that the directions given on Wazuh documentation do not include creation of this file. I’ve found it on your Github site and after adding was able to successfully start the Logstash service which then passed the confgtest.

I will not see if this will start causing logs to be visible in Kibana.

**Comment by snaow** [comment]
If you are installing Elastic 2.0:

http://documentation.wazuh.com/en/latest/ossec_elk_logstash.html#logstash-server-configuration

Quoting:

"sudo cp
> ~/ossec_tmp/ossec-wazuh/extensions/elasticsearch/elastic-ossec-template.json
> /etc/logstash/"


 If your Logstash configuration refers to
"/etc/logstash/wazuh-elastic2-template.json"
is because you are not following strictly the documentation.

Installation / docs summary

Wazuh 1.1.1 + Elastic 2.0: http://documentation.wazuh.com
Wazuh 2.0 (new version, pending to be released) + Elastic 5.0:
https://documentation-dev.wazuh.com/
Wazuh 2.0 + Elastic 2.0 (not support for Wazuh App):  There is no specific
documentation for installing this setup from the scratch.

As the documentation said (both of them), remember to add Logstash user to
OSSEC group:

usermod -a -G ossec logstash


Regards,
Pedro.

**Comment by marcjbaker** [comment]
Pedro,

Is there another person I possible could communicate with. It seems like you are unsure of how to get the data forwarded from the alerts.json file. Your first recommendation was to install Docker which added nothing to the logstash configuration. The link in this last email is for a forwarder and we are using a single node. Prior to this you directed me to configure the indices per online documentation which caused the service to fail. The Github site has a single node config file which you seem to be unaware of and at this point I feel like I am being followed by a blind man who has as much of an idea as I do about the configuration.

Currently Elasticsearch, Logstash, and Kibana services are running. The OSSEC fields have been mapped and OSSEC dashboards imported but the alerts.json file is not being read or being forwarded to Kibana. It seems like a simple configuration issue but as this is my first time working through the system it would be helpful to have guidance as the Wazuh online documentation does not seem to be complete.

Thanks you.

Marc Baker

**Comment by snaow** [comment]
Hi Marc, 

Feel free to bring this conversation to Wazuh mailing list, some other people could help you there.

> https://groups.google.com/forum/#!forum/wazuh

Pedro.

**Comment by marcjbaker** [comment]
Thank you. I have applied for permission to post. Apologize if I sound frustrated but this has taken hours more than it should have due to the incomplete documentation. We now have Logstash, Kibana, and Elasticstash services running but no data in Kibana. The configuration for a single server node was finally found (not on the Wazuh site or Wazzuh Hids site but on another Git Hub site. We copied as directed into /etc/logstash/conf.d/ restarted the service but still no data appearing in Kibana. You originally said this was due to our not installing Docker and when we did with no changes you forwarded another suggestion which was followed by a third when that did not work. At this point I will post on the Google group because the visualizations and dashboards in Kibana look promising if it is possible to actually get data into it. The problem is that your online documentation has been gone through numerous times with no answer in site and as said previously, you seem to be guessing about the answer as much as I am.

V/r

Marc
